assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java

```diff
@@ -11557,7 +11607,15 @@ public void startElement(String uri, String localName, String qName,
 							m_feedbackText = formattedTextDecodeFormattedTextAttribute(attributes, "feedbacktext");
 
 							// SAK-17606
-							m_anonymousSubmissionId = m_id.substring(27)+" (" + rb.getString("grading.anonymous.title")  + ")";
+							String id = m_id.substring(27);
+							try {
+								Site site = SiteService.getSite(m_context);
+								if(candidateDetailProvider.useInstitutionalAnonymousId(site)) {
+									id = candidateDetailProvider.getCandidateID(UserDirectoryService.getUser(attributes.getValue("submitterid")), site).orElse(m_id.substring(27));
+								}
+							} catch(Exception ignore){}
```

Comment:
Can we not just catch all `Exception`s, just the Exceptions that we're expecting to see.